### PR TITLE
chore: Bump Buf to v1.32.0 and move to v2

### PR DESCRIPTION
**Pull request by codingllama**
Update to the latest release and migrate configs to v2.

* https://github.com/bufbuild/buf/releases/tag/v1.32.0

### CODE REVIEW

File: api/proto/buf.lock [hunk 1]
```diff
@@ -4,8 +4,10 @@ deps:
   - remote: buf.build
     owner: gogo
     repository: protobuf
-    commit: b03c65ea87cdc3521ede29f62fe3ce239267c1bc
```
Comment (codingllama) [inline api/proto/buf.lock:7]: Buf v1.32.0 is really unhappy about this hash. I ended up deleting the lock files and creating them again with `buf dep update`.

AFAICT this is still the same underlying commit, except the hash is from the Buf registry: https://buf.build/gogo/protobuf/commits/commit/4df00b267f944190a229ce3695781e99. If you look under "associated labels" you can see b03c65ea87cdc3521ede29f62fe3ce239267c1bc in there.

File: gen/proto/ts/google/protobuf/descriptor_pb.ts [hunk 2]
```diff
@@ -394,12 +394,12 @@ export interface FieldDescriptorProto {
      * If true, this is a proto3 "optional". When a proto3 field is optional, it
      * tracks presence regardless of field type.
      *
-     * When proto3_optional is true, this field must be belong to a oneof to
-     * signal to old proto3 clients that presence is tracked for this field. This
-     * oneof is known as a "synthetic" oneof, and this field must be its sole
-     * member (each proto3 optional field gets its own synthetic oneof). Synthetic
-     * oneofs exist in the descriptor only, and do not generate any API. Synthetic
-     * oneofs must be ordered after all "real" oneofs.
+     * When proto3_optional is true, this field must belong to a oneof to signal
```
Comment (codingllama) [inline gen/proto/ts/google/protobuf/descriptor_pb.ts:397]: Unclear to me why this changed now, but it seems to be mostly edition related.

File: proto/buf.yaml [hunk 3]
```diff
@@ -1,7 +1,4 @@
 version: v1
-deps:
```
Comment (codingllama) [inline proto/buf.yaml:2]: This dep is not in use so I just took it out.

**Comment by codingllama** [comment]
I do have a draft for a v2 config, but it's giving me trouble over a `google/rpc/status.proto` import. 🤷

**Comment by codingllama** [comment]
> I do have a draft for a v2 config, but it's giving me trouble over a `google/rpc/status.proto` import. 🤷

I got it. I'll mail something Monday.

**Comment by codingllama** [comment]
PTAL - migrated to v2 config. It actually cuts some duplication in the linter declarations.

**Comment by codingllama** [comment]
Buf breaking is mad because it wants 32-char lock commits on the master branch too. It looks like we can push that even with Buf v1.31, so I've done that at #41776.

**Review by rosstimothy — APPROVED** [review]

**Review by gabrielcorado — APPROVED** [review]

**Pull request merged**